lib/spack/spack/cmd/checkbinaries.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from __future__ import print_function
+
+from six.moves.urllib.request import urlopen
+from six.moves.urllib.error import URLError
+
+import json
+import os
+import re
+import yaml
+
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.config
+from spack.error import SpackError
+from spack.spec import Spec
+from spack.util.executable import which
+
+
+description = "check if binaries need to be rebuilt"
+section = "developer"
+level = "long"
+
+
+_buildcache_path = os.path.join('build_cache')
+_pkg_match_regex = re.compile(r'packages[/\\]([^/\\]+)[/\\]package\.py')
+_pkg_yaml_match_regex = re.compile(r'<a href=[^>]+\>\s*([^<]+)\s*\<')
+
+
+def changed_package_files(args):
+    """Get list of changed package.py files in the Spack repository."""
+
+    git = which('git', required=True)
+
+    range = "{0}...".format(args.base)
+
+    git_args = [
+        # Add changed files committed since branching off of develop
+        ['diff', '--name-only', '--diff-filter=ACMR', range],
+        # Add changed files that have been staged but not yet committed
+        ['diff', '--name-only', '--diff-filter=ACMR', '--cached'],
+        # Add changed files that are unstaged
+        ['diff', '--name-only', '--diff-filter=ACMR'],
+    ]
+
+    changed = set()
+
+    for arg_list in git_args:
+        files = git(*arg_list, output=str).split('\n')
+
+        for f in files:
+            # Look only for modified package.py files
+            m = _pkg_match_regex.search(f)
+            if m:
+                changed.add(m.group(1))
+
+    return sorted(changed)
+
+
+def compute_local_hash(pkg_name, spec_yaml):
+    local_spec = Spec.from_yaml(spec_yaml)
+    tty.msg('local dag hash for %s: %s' % (pkg_name, local_spec.dag_hash()))
```

Comment:
the parsed `Spec` should have a `name` attribute already, so you probably don't need the `pkg_name` parameter.